sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs

```diff
@@ -82,17 +82,35 @@ namespace Azure.Storage.Blobs.Specialized
     /// </summary>
     public class BlockBlobClient : BlobBaseClient
     {
+        /// <summary>
+        /// Gets the maximum number of bytes that can be sent in a call
+        /// to <see cref="UploadAsync(Stream, UploadBlobOptions, CancellationToken)"/>. Supported value is now larger
+        /// than <see cref="int.MaxValue"/>; please use
+        /// <see cref="BlockBlobMaxUploadBlobBytesLong"/>.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public virtual int BlockBlobMaxUploadBlobBytes => int.MaxValue;
```

Comment:
When I tried to do this in Java I got "API compatibility error". I guess it's not happening here because we alias constant via property. Java error made me think that we probably shouldn't be doing that - let existing code do the same thing in case someone is using this value in some calculations.

Reply:
I don't understand the benefit to leaving this value back at 256 MB.

Reply:
It's more of - what can break if we move publicly available "constant" that someone could take dependency on. Someone may come and say that after upgrade their throughput have changed without them making any change on their side.

Reply:
If that is our mindset then I don't think we should expose a new one, keeping all our constants internal and marking this old one `EditorBrowsable.Never`. If we can't change a value that is designed to communicate the current max value, which we know for certain will change again in the future, then we shouldn't be making publicly available values for the current max that we have to keep hiding and exposing a new one for.

Reply:
I got stopped by static analysis in Java when I tried to do the same there - it told me it's breaking API change. Therefore I'd check .NET SDK guidelines related to this change.

On the other hand - if we can't do this in Java - should we in .NET?

And then... Python doesn't validate that at all...

@tg-msft could you chime in here?

Reply:
We made these properties in .NET instead of constants specifically so we could version them when this happened.  I think we should make this change for .NET even if we can't for Java _(but start using properties for new constants in Java as well so we'll have that wiggle room if we ever need it)_.

This is one of those things that we should really consider checking the version for and return 256 MB for everything less than the current and `int.MaxValue` from here on out.  The original suggestion was to only add version checks when there was a potential for data corruption or loss, but anyone doing a check on this constant to upload with the old service version will have a very unpleasant time.  It would be as as easy as `public virtual int BlockBlobMaxUploadBlobBytes => Version < BlobClientOptions.ServiceVersion.Vxyz ? 256 * Constants.MB : int.MaxValue;`